client/verta/tests/test_utils/test_http_session_utils.py

```diff
@@ -0,0 +1,110 @@
+# -*- coding: utf-8 -*-
+""" Unit tests for the request session utilities found in verta._internal_utils.http_session_utils"""
+
+import pytest
+import requests
+
+from requests import Session
+from requests.adapters import HTTPAdapter
+from requests.cookies import RequestsCookieJar
+from typing import OrderedDict
+from unittest.mock import patch
+from urllib3 import Retry
+
+from verta._internal_utils import http_session_utils
+
+# Retry object for patching
+MOCK_RETRY_OBJECT: Retry = http_session_utils.retry_config(max_retries=11,
+                                            status_forcelist={123, 456},
+                                            backoff_factor=1.23
+                                            )
+
+# A different retry object for testing on-the-fly alteration
+ALTERED_MOCK_RETRY_OBJECT: Retry = http_session_utils.retry_config(max_retries=22,
+                                                           status_forcelist={789},
+                                                           backoff_factor=2.45
+                                                           )
+# HTTPAdapter object for patching
+MOCK_HTTP_ADAPTER = HTTPAdapter(max_retries=MOCK_RETRY_OBJECT)
```

Comment:
Nit: should we use pytest fixtures instead of global variables? Global variables assumes they don't have state that will make the tests coupled.

Reply:
I would have strongly preferred to use fixtures here, but python throws an error if you try to use a purest.fixture as the return value of a patch.  A global variable allowed me to inject a singular test wherever I wanted without jumping through extra hoops to work around that limitation.  Perhaps we can explore this further

Reply:
Interesting. Maybe a comment with such note here might help bc I'm sure I'll forget it in the future